server/src/main/java/org/apache/iotdb/db/rescon/PrimitiveArrayManager.java

```diff
@@ -266,7 +266,7 @@ public static void close() {
    * @return an array of primitive data arrays
    */
   public static Object createDataListsByType(TSDataType dataType, int size) {
-    int arrayNumber = (int) Math.ceil((float) size / (float) ARRAY_SIZE);
+    int arrayNumber = size / ARRAY_SIZE + 1;
```

Comment:
如果size可以整除ARRAY_SIZE，后边的+1似乎是多余的
```suggestion
    int arrayNumber = size / ARRAY_SIZE + (size % ARRAY_SIZE == 0 ? 0 : 1);
```